tests/integration/custom-authentication/customPlugin.js

```diff
@@ -0,0 +1,43 @@
+class CustomPlugin {
+  constructor(serverless, options) {
+    this.serverless = serverless
+    this.options = options
+
+    this.hooks = {
+      initialize: () => {
+        this.loadPlugin()
+      },
+    }
+  }
+
+  loadPlugin() {
+    const authProvider = {
+      // eslint-disable-next-line no-unused-vars
+      getAuthorizationStrategy: (endpoint, functionKey, method, path) => {
+        return {
+          name: 'strategy-name',
+          scheme: 'scheme',
+
+          getAuthenticateFunction: () => ({
+            async authenticate(request, h) {
+              const context = { expected: 'it works' }
+              return h.authenticated({
+                credentials: {
+                  context,
+                },
+              })
+            },
+          }),
+        }
+      },
+    }
+
+    this.options.plugins = {
```

Comment:
Doing that will change globally `options` object for all plugins and serverless internals.

Any modifications to it are already coined as an anti-pattern: https://github.com/serverless/serverless/issues/2582

It would be nicer to attach this configuration not via plugin, but simply via external module, something close to:

_serverless.yml_
```yaml
custom:
  offline:
    customAuthorizationStrategy: ./authorization-strategy.js
```

_authorization-strategy.js_
```js
module.exports =  (endpoint, functionKey, method, path) => {
   ...
}
```

Reply:
@medikoo Thanks for the advice and apologies for the late reply.  I believe I understand your suggestion, I just need a bit of time to look more into how custom variables are managed in serverless and see how it applies to injecting this behavior from an external package.

Internally we've been using the changes represented in this branch for all our projects with serverless-offline.  With that, other items of work have been more pressing to dedicate time to.  It will be a few more days before I jump back into this to make the suggested changes but given how quickly time has gone by since your last response I wanted to let you know that there is still an intention to continue forward with this.

Reply:
@medikoo Thanks for the advice and apologies for the late reply.  I believe I understand your suggestion,  I need a bit of time to look into how serverless custom attributes are managed and how to inject the desired behavior from an external package.

Time has gone by quickly since your last reply and I wanted you to know that the intention is still to follow through with the suggested changes.  It will be a few more days before I get a chance to get this updated.

Thanks again

Reply:
Great thanks @tom-stclair for heads up. No worries, we look forward the update!

Reply:
@medikoo - had a chance to make the suggested changes.  Let me know if you had something else in mind.